tests/lib/box-sdk-node-test.js

```diff
@@ -131,6 +160,56 @@ describe('box-node-sdk', function() {
 		});
 	});
 
+	describe('getPreconfiguredInstance()', function() {
+		it('should set config for all passed in params when called', function() {
+
+			sdk = BoxSDKNode.getPreconfiguredInstance(TEST_APP_SETTINGS);
+
+			assert.ok(TokenManagerConstructorStub.calledWithNew(), 'Should construct new TokenManager');
+			assert.ok(TokenManagerConstructorStub.calledWithMatch(TEST_APP_SETTINGS_CONFIG), 'TokenManager should be passed config');
+			assert.ok(sdk, 'SDK should be constructed');
+		});
+
+		it('should create an anonymous session with config when called', function() {
+
+			sdk = BoxSDKNode.getPreconfiguredInstance(TEST_APP_SETTINGS);
+
+			assert.ok(AnonymousAPISession.calledWithNew(), 'Should construct new anonymous session');
+			assert.ok(AnonymousAPISession.calledWithMatch(TEST_APP_SETTINGS_CONFIG), 'Anonymous session should be passed config');
+			assert.ok(sdk, 'SDK should be constructed');
+		});
+
+		it('should create an API Request Manager with config when called', function() {
+
+			sdk = BoxSDKNode.getPreconfiguredInstance(TEST_APP_SETTINGS);
+
+			assert.ok(APIRequestManagerConstructorStub.calledWithNew(), 'Should construct new APIRequestManager');
+			assert.ok(APIRequestManagerConstructorStub.calledWithMatch(TEST_APP_SETTINGS_CONFIG), 'APIRequestManager should be passed config');
+			assert.instanceOf(APIRequestManagerConstructorStub.getCall(0).args[1], EventEmitter, 'APIRequestManager should be passed event bus');
+			assert.ok(sdk, 'SDK should be constructed');
+		});
+	});
+
+	describe('configure()', function() {
+		beforeEach(function() {
+			sdk = BoxSDKNode.getPreconfiguredInstance(TEST_APP_SETTINGS);
+		});
+
+		it('should verify that additional parameters can be passed to the BoxSDKNode instance', function() {
+			var additonalParams = {
+				apiRootURL: 'myUrl',
+				retryIntervalMS: 11111,
+				numMaxRetries: 3
+			};
+
+			sdk.configure(additonalParams);
+			assert.equal(sdk.config.clientID, 'myId');
+			assert.equal(sdk.config.apiRootURL, 'myUrl');
```

Comment:
Assertions like this are easier if you use something like `assert.deepPropertyVal(sdk, 'config.apiRootURL', 'myUrl')`, which provides better error messaging if any of the intermediate objects is not present.